diff --git a/mppsolar/outputs/prom_push.py b/mppsolar/outputs/prom_push.py
index f10ecdd..885d8fc 100644
--- a/mppsolar/outputs/prom_push.py
+++ b/mppsolar/outputs/prom_push.py
@@ -1,10 +1,11 @@
 import logging
+import socket
 
 try:
     import requests
 except ImportError:
     print("You are missing dependencies in order to be able to use that output.")
-    print("To install them, use that command:")
+    print("To install them, use this command:")
     print("    python -m pip install 'mppsolar[push]'")
     requests = None
 
@@ -16,20 +17,34 @@ log = logging.getLogger("prom")
 
 class prom_push(prom):
     push_url = ""
+    job = "mppsolar"  # static default
 
     def __str__(self):
-        return "pushes Node exporter Prometheus format to PushGateway"
+        return "Pushes Prometheus exposition format directly to VictoiaMetrics (No Gateway)"
 
     def output(self, *args, **kwargs):
-        # We override the method only to get the PushGateway URL from the config
-        self.push_url = kwargs["push_url"]
+        log.debug(f"RAW kwargs: {kwargs}")
 
+        self.push_url = kwargs.get("push_url")
+        self.job = kwargs.get("job", "mppsolar")
+
+        # Fully ignore provided instance, always sanitize from hostname
+        self.instance = socket.gethostname().split('.')[0]
+
+        log.debug(f"Forced instance to short hostname: {self.instance}")
         return super().output(*args, **kwargs)
 
     def handle_output(self, content: str) -> None:
         if not requests:
             return
 
-        with requests.post(self.push_url, data=content) as req:
-            log.debug(f"POST'ed data to PushGateway {self.push_url!r}: {req=}")
+        headers = {'Content-Type': 'text/plain'}
+        target_url = f"{self.push_url}/metrics/job/{self.job}/instance/{self.instance}"
+
+        try:
+            with requests.post(target_url, data=content, headers=headers) as req:
+                log.debug(f"POST'ed data to PushGateway {target_url!r}: status_code={req.status_code}")
+        except requests.RequestException as e:
+            log.error(f"Failed to push to PushGateway: {e}")
+
     
\ No newline at end of file
